econml/sklearn_extensions/model_selection.py

```diff
@@ -578,18 +604,19 @@ def __init__(self, models, unwrap=True):
         self.models = [clone(model, safe=False) for model in models]
         self.unwrap = unwrap
 
-    def train(self, is_selecting, *args, **kwargs):
+    def train(self, is_selecting, folds: Optional[List], *args, **kwargs):
+        assert len(self.models) > 0, "ListSelector must have at least one model"
         if is_selecting:
             scores = []
             for model in self.models:
-                model.train(is_selecting, *args, **kwargs)
+                model.train(is_selecting, folds, *args, **kwargs)
                 scores.append(model.best_score)
             self._all_scores = scores
             self._best_score = np.max(scores)
             self._best_model = self.models[np.argmax(scores)]
 
         else:
-            self._best_model.train(is_selecting, *args, **kwargs)
+            self._best_model.train(is_selecting, folds, *args, **kwargs)
```

Comment:
`is_selecting` will be False here. Do we need to pass `folds` in this case?

Reply:
It's academic; if `is_selecting` is False then `folds` will be None anyway so we could pass `folds` or we could equivalently pass `None`.

But I think this is slightly more robust to changes in the upstream logic, since if for some reason we ever change that invariant then the logic here might not need to be changed.